mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +349,37 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _infer_artifact_type_and_ext(artifact_name, raw_artifact, exception_header):
```

Comment:
Not a big fan of passing strings around as method parameters purely for exception content. Can we instead pass the custom metric and index (ideally these could be represented as a `namedtuple` as well) as a parameter to this method and use them to construct the exception content?